src/Raven.Client/Http/NodeSelector.cs

```diff
@@ -66,14 +66,18 @@ public bool OnUpdateTopology(Topology topology, bool forceUpdate = false)
             var state = _state;
             var stateFailures = state.Failures;
             var serverNodes = state.Nodes;
-            var len = Math.Min(serverNodes.Count, stateFailures.Length);
+            var numOfServers = serverNodes.Count;
+            var len = Math.Min(numOfServers, stateFailures.Length);
```

Comment:
what if I have 3 servers, but I have 1 failure (node A), and I want node C?

Reply:
- A throws an exception
- We will get back here _if (serverNodes[i].ClusterTag == nodeTag)_  again, Let's assume C has no failures so we will return him, no?

Reply:
You will scan just the first item, since `stateFailures.Length == 1` in this senario, no?

Reply:
Just a question, Any specific reason to use `Math.Min` rather than using the length of the nodes?
This doesn't simplify things ?
```
var state = _state;
var stateFailures = state.Failures;
var serverNodes = state.Nodes;
var numOfServers = serverNodes.Count;
for (var i = 0; i < numOfServers; i++)
{
    if (serverNodes[i].ClusterTag == nodeTag)
    {
        if (numOfServers == 1)
            return (i, serverNodes[i]);
        
        if (stateFailures[i] == 0)
            return (i, serverNodes[i]);
    
        throw new RequestedNodeUnavailableException($"Requested node {nodeTag} currently unavailable, please try again later.");
    }
}
```

Reply:
Is the `stateFailures` length and the `serverNodes` length the same always?

Reply:
https://github.com/ravendb/ravendb/blob/df55b8c99ceac9069dcb8b8fa9e154f9ce9521dd/src/Raven.Client/Http/NodeSelector.cs#L24

This is sufficient, no?

Reply:
So there are always going to be the same size, then?
So why the `Min` ?

Reply:
@grisha-kotler 
https://github.com/ravendb/ravendb/blob/df55b8c99ceac9069dcb8b8fa9e154f9ce9521dd/src/Raven.Client/Documents/Changes/DatabaseChanges.cs#L491

To me, It make sense to use the Min within _GetPreferredNode()_ rather than _GetRequestedNode(nodeTag)_
We will miss Oren's use case discussed above, no?

Reply:
ignore